torch/_inductor/compile_fx.py

```diff
@@ -436,37 +436,33 @@ def get_patched_config_dict(config_patches=None) -> Dict[str, Any]:
         return config.get_config_copy()
 
 
-def with_fresh_cache_if_config(fn):
-    @functools.wraps(fn)
-    def wrapper(*args, **kwargs):
-        if config.force_disable_caches:
-            # Don't delete the cache dir because it has to survive beyond the
-            # compile_fx call. Let's put the temp dirs under the default cache
-            # dir so they're easier to locate.
-            with fresh_inductor_cache(dir=cache_dir(), delete=False):
-                return fn(*args, **kwargs)
-        else:
-            return fn(*args, **kwargs)
-
-    return wrapper
+@contextlib.contextmanager
```

Comment:
wouldn't the "wrapper" created in here defeat the purpose of making it a context manager?

Reply:
Uhhh, no? There's no more wrapper being created? Unless you're suggesting the application of the context manager should be lazy? But I'm pretty sure it is not...